atomic_reactor/plugins/pre_add_filesystem.py

```diff
@@ -0,0 +1,209 @@
+        # Unable to re-use koji's code because "cli" is not
+        # a package of koji and this logic is intermingled
+        # with CLI specific instructions.
+
+        args = []
+        opts = {}
+
+        config = ConfigParser()
+        config.read(config_file_name)
+
+        image_name = None
+
+        section = 'image-build'
+        for option in ('name', 'version', 'arches', 'target', 'install_tree'):
+            value = config.get(section, option)
+            if option == 'arches':
+                value = [arch for arch in value.split(',') if arch]
+            elif option == 'name':
+                image_name = value
+            args.append(value)
+            config.remove_option(section, option)
+
+        for option, value in config.items(section):
+            if option in ('repo', 'format'):
+                value = [v for v in value.split(',') if v]
+            elif option in ('disk_size'):
+                value = int(value)
+            opts[option] = value
+
+        section = 'ova-options'
+        if config.has_section(section):
+            ova = []
+            for k, v in config.items(section):
+                ova.append('{}={}'.format(k, v))
+            opts['ova_option'] = ova
+
+        section = 'factory-parameters'
+        if config.has_section(section):
+            factory = []
+            for option, value in config.items(section):
+                factory.append((option, value))
+            opts['factory_parameter'] = factory
+
+        # Set some defaults.
+        opts.setdefault('disk_size', 10)
+
+        return image_name, args, {'opts': opts}
+
+    def build_filesystem(self):
+        # Kickstart file should be in the same folder as Dockerfile
+        df_path, df_dir = self.workflow.source.get_dockerfile_path()
+        kickstart = os.path.join(df_dir, 'kickstart.conf')
+        if not os.path.exists(kickstart):
+            raise RuntimeError('Kickstart file not found: {}'
+                               .format(kickstart))
+
+        image_name, args, kwargs = self.parse_image_build_config(kickstart)
+        pattern = ('{}.*(\.tar|\.tar\.gz|\.tgz|\.bzip|\.tar\.xz|\.txz)$'
+                   .format(image_name))
+        filesystem_regex = re.compile(pattern, re.IGNORECASE)
+        if self.from_task_id:
+            task_id = self.from_task_id
+        else:
+            task_id = self.session.buildImageOz(*args, **kwargs)
+        return task_id, filesystem_regex
+
+    def find_filesystem(self, task_id, filesystem_regex):
+        for f in self.session.listTaskOutput(task_id):
+            f = f.strip()
+            match = filesystem_regex.match(f)
+            if match:
+                return task_id, match.group(0)
+
+        # Not found in this task, search sub tasks
+        for sub_task in self.session.getTaskChildren(task_id):
+            found = self.find_filesystem(sub_task['id'], filesystem_regex)
+            if found:
```

Comment:
Since we're using `None` as the error case, I think I'd rather change the test to `if found is not None:` but it's not a blocker for merging this.